jobserver/pipeline_config.py

```diff
@@ -46,6 +46,15 @@ def get_actions(config):
         yield {"name": "run_all", "needs": all_actions}
 
 
+def get_database_actions(config):
+    """
+    Return the names of actions that use the backend database.
+    """
+    for name, action in config.actions.items():
+        if action.is_database_action:
+            yield name
```

Comment:
Probably not worth it for this PR but I wonder if it would be useful for pipeline to also grow `Config.database_actions` that's just a property returning this list 🤔  do we need to know the same list in job-runner?  Could be a useful deck-scrubbing task if so…

Reply:
I thought about adding that, but at the time I was thinking of individual actions being failed. But yes, I think job-runner will need to know the same list, especially if it's just going to fail the whole job request.  I'll leave it for now though :)